### PR TITLE
ddl: fix panic in drop column

### PR DESCRIPTION
**Pull request by jackysp**
Drop column with pk-is-handle table (t) when insert into t will cause index out of range in delete only state. PTAL @zimulala @coocood @winkyao . Back trace as below:
```
 2018/02/14 16:54:01.082 ddl_worker.go:51: [error] ddlWorker runtime error: index out of range goroutine 121 [running]:
github.com/pingcap/tidb/util.GetStack(0xc420df8f60, 0x1d09260, 0x28bd4d0)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/util/misc.go:52 +0x74
github.com/pingcap/tidb/ddl.(*ddl).onDDLWorker.func1()
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/ddl/ddl_worker.go:50 +0x6e
panic(0x1d09260, 0x28bd4d0)
	/usr/local/Cellar/go/1.9.3/libexec/src/runtime/panic.go:491 +0x283
github.com/pingcap/tidb/expression.TableInfo2SchemaWithDBName(0x1eb69a1, 0xd, 0x1eb69a1, 0xd, 0xc421350540, 0xc420df92c0)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/expression/expression.go:277 +0xa3a
github.com/pingcap/tidb/plan.(*planBuilder).buildInsert(0xc4212ee0c0, 0xc421368140, 0x0, 0x1e30700)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/plan/planbuilder.go:765 +0xbf
github.com/pingcap/tidb/plan.(*planBuilder).build(0xc4212ee0c0, 0x26a1aa0, 0xc421368140, 0x0, 0xc421398240)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/plan/planbuilder.go:188 +0x26e
github.com/pingcap/tidb/plan.Optimize(0x26c8380, 0xc42126a820, 0x26a1aa0, 0xc421368140, 0x26b3c60, 0xc421366a50, 0x0, 0x0, 0x0, 0xc420df9a88)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/plan/optimizer.go:68 +0x11e
github.com/pingcap/tidb/executor.(*Compiler).Compile(0xc420df9c88, 0x41aa370, 0xc4200160c0, 0x26aa8a0, 0xc421368140, 0x0, 0x0, 0x0)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/executor/compiler.go:48 +0x1eb
github.com/pingcap/tidb.(*session).Execute(0xc42126a820, 0x41aa370, 0xc4200160c0, 0x1ee79ca, 0x38, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/session.go:807 +0xad4
github.com/pingcap/tidb/ddl_test.(*testStateChangeSuite).runTestInSchemaState.func1(0xc4212e8f20)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/ddl/ddl_db_change_test.go:339 +0x177
github.com/pingcap/tidb/ddl.(*TestDDLCallback).OnJobUpdated(0xc42126de30, 0xc4212e8f20)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/ddl/callback_test.go:43 +0x64
github.com/pingcap/tidb/ddl.(*ddl).handleDDLJobQueue(0xc420d08000, 0x1, 0x0)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/ddl/ddl_worker.go:252 +0x159
github.com/pingcap/tidb/ddl.(*ddl).onDDLWorker(0xc420d08000)
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/ddl/ddl_worker.go:64 +0x224
created by github.com/pingcap/tidb/ddl.(*ddl).start
	/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/ddl/ddl.go:318 +0x13a
```

### CODE REVIEW

**Comment by jackysp** [comment]
@GregoryIan this could help for your tests.

**Comment by jackysp** [comment]
CI fixed. PTAL @shenli

File: ddl/column.go [hunk 1]
```diff
@@ -32,48 +32,63 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-// adjustColumnInfo is used to set the correct position of column info.
+// TODO: Index can't cover the add column with offset now, we may check this later.
+// adjustColumnInfoInAddColumn is used to set the correct position of column info when add column.
```
Comment (shenli) [inline ddl/column.go]: when adding

File: ddl/column.go [hunk 2]
```diff
@@ -32,48 +32,63 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-// adjustColumnInfo is used to set the correct position of column info.
+// TODO: Index can't cover the add column with offset now, we may check this later.
```
Comment (shenli) [inline ddl/column.go]: Can not understand the TODO.

File: session_test.go [hunk 3]
```diff
@@ -1870,7 +1870,6 @@ func (s *testSessionSuite) TestCastTimeToDate(c *C) {
 }
 
 func (s *testSessionSuite) TestSetGlobalTZ(c *C) {
-	defer testleak.AfterTest(c)()
```
Comment (shenli) [inline session_test.go]: Why remove this?

File: ddl/column.go (hunk 2 above)
Comment (jackysp) [reply ddl/column.go]: Just move it from L66, it means to adjust the index column offset, need a double check.

File: session_test.go (hunk 3 above)
Comment (jackysp) [reply session_test.go]: This line will cause an error in leak test, all the cases should be checked in Setup and Teardown in TestSuite.

**Comment by jackysp** [comment]
PTAL @shenli

File: ddl/column.go (hunk 2 above)
Comment (coocood) [reply ddl/column.go]: We only need to do it when we support adding column and index at the same time.

Comment (coocood) [reply ddl/column.go]: How about:
```
TODO: this implementation doesn't support adding column and index at the same time.
```

Comment (jackysp) [reply ddl/column.go]: In this function, it updates the offsets of the index columns. I think it is not about adding an index.

Comment (jackysp) [reply ddl/column.go]: Make it more readable, PTAL @coocood .

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**